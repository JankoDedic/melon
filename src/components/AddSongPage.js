import React from 'react';
import { connect } from 'react-redux';

import SongForm from './SongForm';
import { startAddSong } from '../actions/songs';

class AddSongPage extends React.Component {
  handleSubmit = (song) => {
    this.props.startAddSong(song);
    this.props.history.push('/dashboard');
  }
  render() {
    return (
      <SongForm formTitle="Add Song" onSubmit={this.handleSubmit} />
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  startAddSong: (song) => dispatch(startAddSong(song)),
});

export default connect(undefined, mapDispatchToProps)(AddSongPage);
