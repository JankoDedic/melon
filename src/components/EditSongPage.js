import React from 'react';
import { connect } from 'react-redux';

import SongForm from './SongForm';
import { startEditSong, removeSong } from '../actions/songs';

class EditSongPage extends React.Component {
  handleSubmit = (song) => {
    this.props.startEditSong(this.props.song.id, song);
    this.props.history.push('/dashboard');
  }
  handleRemove = () => {
    this.props.removeSong(this.props.song.id);
    this.props.history.push('/dashboard');
  }
  render() {
    return (
      <div>
        <SongForm
          formTitle="Edit Song"
          onSubmit={this.handleSubmit}
          song={this.props.song}
        />
        <button
          className="form-button"
          onClick={this.handleRemove}
        >
          Remove Song
        </button>
      </div>
    );
  }
}

const mapStateToProps = (state, props) => ({
  song: state.songs.find((song) => song.id === props.match.params.id),
});

const mapDispatchToProps = (dispatch, ownProps) => ({
  startEditSong: (id, updates) => dispatch(startEditSong(id, updates)),
  removeSong: (id) => dispatch(removeSong(id)),
});

export default connect(mapStateToProps, mapDispatchToProps)(EditSongPage);
