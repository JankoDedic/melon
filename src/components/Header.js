import React from 'react';

export default () => (
  <div className="header-container">
    <div className="header">
      <div className="header__logo">
        Melon
      </div>
      <button className="header__logout">Logout</button>
    </div>
  </div>
);
