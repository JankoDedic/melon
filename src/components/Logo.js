import React from 'react';

export const Logo = ({ className }) => {
  let imageClassNames;
  if (className) {
    imageClassNames = className.split(' ').map(name => name + '__image').join(' ');
  }
  return (
    <div className={className}>
      <div className={imageClassNames}></div>
      melon
    </div>
  );
};

export default Logo;
