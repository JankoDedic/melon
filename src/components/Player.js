import React from 'react';
import SeekBar from './SeekBar';
import VolumeSlider from './VolumeSlider';

class Player extends React.Component {
  state = {
    isPlaybackActive: false,
  };

  togglePlayback = () => {
    this.setState((prevState) => ({
      isPlaybackActive: !prevState.isPlaybackActive
    }));
  };

  render() {
    return (
      <div>
        <button onClick={this.togglePlayback}>play/pause</button>
        <SeekBar
          duration={60}
          isPlaybackActive={this.state.isPlaybackActive}
          onSeek={(value) => { console.log('You seeked to', value); }}
          onPlaybackPause={() => { console.log('Playback paused'); }}
          onPlaybackResume={() => { console.log('Playback resumed'); }}
        />
        <VolumeSlider />
      </div>
    );
  }
}

export default Player;
