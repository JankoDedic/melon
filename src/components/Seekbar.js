import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Slider from '@material-ui/lab/Slider';

const styles = {
  root: {
    width: '100%',
  },
};

class Seekbar extends React.Component {
  state = {
    value: 0,
    isHandleDragged: false,
    canSeekOnDragEnd: false,
  };

  // NOTE: Slider component sometimes calls onDragStart twice in cases when
  // the click is near the handle. This should not matter in practice, but
  // you should be aware that you should not rely on onHandleDragStart to
  // perform any toggling action.
  onDragStart = () => {
    this.setState((prevState, props) => ({
      isHandleDragged: true,
      value: props.progress
    }));

    this.props.onHandleDragStart();
  }

  // A click will call onDragEnd -> onHandleChange
  // A drag and release will call onHandleChange -> onDragEnd
  // - onHandleChange is free to call onSeek unless the handle is still being
  // dragged (in that case, the responsibility is delegated to onDragEnd)
  // - onDragEnd is allowed to call onSeek only if onHandleChange allowed it
  // (via canSeekOnDragEnd boolean).

  onDragEnd = () => {
    this.props.onHandleDragEnd();

    let canSeekOnDragEnd = this.state.canSeekOnDragEnd;

    if (this.state.canSeekOnDragEnd) {
      this.props.onSeek(this.state.value);
      canSeekOnDragEnd = false;
    }

    this.setState({ isHandleDragged: false, canSeekOnDragEnd });
  }

  onHandleChange = (e, value) => {
    let canSeekOnDragEnd;

    if (this.state.isHandleDragged) {
      canSeekOnDragEnd = true;
    } else {
      this.props.onSeek(value);
    }

    this.setState({ value, canSeekOnDragEnd });
  }

  render() {
    const { classes, progress } = this.props;
    const { value, isHandleDragged } = this.state;

    return (
      <div className={classes.root}>
        <Slider
          max={this.props.duration}
          value={isHandleDragged ? value : progress}
          aria-labelledby="label"
          onChange={this.onHandleChange}
          onDragStart={this.onDragStart}
          onDragEnd={this.onDragEnd}
        />
      </div>
    )
  }
}

Seekbar.propTypes = {
  duration: PropTypes.number.isRequired,
  progress: PropTypes.number.isRequired,

  onSeek: PropTypes.func,
  onHandleDragStart: PropTypes.func,
  onHandleDragEnd: PropTypes.func,
};

Seekbar.defaultProps = {
  onSeek: () => {},
  onHandleDragStart: () => {},
  onHandleDragEnd: () => {},
};

export default withStyles(styles)(Seekbar);
