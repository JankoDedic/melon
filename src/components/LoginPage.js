import React from 'react';
import { connect } from 'react-redux';

import Logo from './Logo';
import { startLogin, startAnonymousLogin } from '../actions/auth';

export const LoginPage = ({ startLogin, startAnonymousLogin }) => (
  <div className="login-page-container">
    <div className="login-page">
      <Logo className="login-page__logo" />
      <div className="login-page__slogan">
        One music collection. Any stream.
      </div>
      <button
        type="button"
        className="login-page__button"
        onClick={startLogin}
      >
      </button>
      <button
        type="button"
        className="login-page__guest-login-button"
        onClick={startAnonymousLogin}
      >
        Login as Guest
      </button>
    </div>
  </div>
);

const mapDispatchToProps = (dispatch) => ({
  startLogin: () => dispatch(startLogin()),
  startAnonymousLogin: () => dispatch(startAnonymousLogin()),
});

export default connect(undefined, mapDispatchToProps)(LoginPage);
