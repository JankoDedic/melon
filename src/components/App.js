import React from 'react';
import { connect } from 'react-redux';

import AppRouter from '../routers/AppRouter';
import Player from './Player';

export const App = ({ isUserAuthenticated }) => (
  <div>
    <AppRouter />
    {isUserAuthenticated && <Player />}
  </div>
);

const mapStateToProps = (state) => ({
  isUserAuthenticated: !!state.auth.uid,
});

export default connect(mapStateToProps)(App);
