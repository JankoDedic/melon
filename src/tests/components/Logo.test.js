import React from 'react';
import { shallow } from 'enzyme';

import { Logo } from '../../components/Logo';

test('Logo gets rendered properly', () => {
  const wrapper = shallow(<Logo />);
  expect(wrapper).toMatchSnapshot();
});
