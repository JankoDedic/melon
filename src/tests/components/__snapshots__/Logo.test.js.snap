// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo gets rendered properly 1`] = `
<div>
  <div />
  melon
</div>
`;
